src/OpenTelemetry/Logs/LogRecord.cs

```diff
@@ -81,21 +81,21 @@ public sealed class LogRecord
 
         public EventId EventId { get; }
 
-        public string FormattedMessage { get; }
+        public string FormattedMessage { get; set; }
 
         /// <summary>
-        /// Gets the raw state attached to the log. Set to <see
+        /// Gets or sets the raw state attached to the log. Set to <see
         /// langword="null"/> when <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled.
         /// </summary>
-        public object State { get; }
+        public object State { get; set; }
 
         /// <summary>
-        /// Gets the parsed state values attached to the log. Set when <see
+        /// Gets or sets the parsed state values attached to the log. Set when <see
         /// cref="OpenTelemetryLoggerOptions.ParseStateValues"/> is enabled
         /// otherwise <see langword="null"/>.
         /// </summary>
-        public IReadOnlyList<KeyValuePair<string, object>> StateValues { get; }
+        public IReadOnlyList<KeyValuePair<string, object>> StateValues { get; set; }
```

Comment:
The current SDK ensures that State and StateValues are never both non-null. With this change, we'll break that behavior. While it may be okay, we need to analyze the implication.
I'd hold off merging this PR for now, until 1.2 stable ship has sailed. (Its too risky to introduce this directly to a stable package without enough bake time.)

Reply:
> The current SDK ensures that State and StateValues are never both non-null. 

I guess what you mean is that the current main version ensures that either State or StateValues will be null.
If that is what you mean, I didn't find the logic to guarantee that. But I do find that State and StateValues is mutually exclusive - if one exists, the other would be null, based on `ParseStateValues` flag.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/5587855a539072736a02f2bf5ef680cbdba2c3e4/src/OpenTelemetry/Logs/OpenTelemetryLogger.cs#L61

So I guess what we need to understand here is -
what if the user uses the exposed setter to set both State and StateValues to a single LogRecord to be both not null?
My answer is that is something the user should take care of, but not the SDK.
If we want to enforce that, there will be perf penalty on the setters path which is not negligible.